src/App.vue

```diff
@@ -1,35 +1,71 @@
 <template>
   <main>
-    <c-header />
-    <c-table :messages="messages" />
+    <c-header
+      @onUpdateMessages="updateMessages"
+      @onUpdateRefreshInterval="updateRefreshInterval"
+      :refreshInterval="refreshInterval"
+    />
+    <c-table :messages="messages" class="pt-32" />
+    <div class="flex w-full">
+      <div class="loader" v-if="isLoading">Loading...</div>
+    </div>
   </main>
 </template>
 
 <script>
 import CHeader from '@/components/CHeader';
 import CTable from '@/components/CTable';
 import api from '@/api';
+require('@/assets/css/spinner.css');
 export default {
   name: 'App',
   components: { CHeader, CTable },
 
   data() {
     return {
-      messages: []
+      intervalId: null,
+      isLoading: false,
+      messages: [],
+      refreshInterval: 30
     };
   },
 
   methods: {
-    updateMessages() {
-      api.getMessages().then(messages => (this.messages = messages));
+    startUpdateMessages() {
+      this.intervalId = setInterval(() => {
+        this.updateMessages();
+      }, this.refreshInterval * 1000);
+    },
+    wait(ms) {
+      return new Promise(resolve => setTimeout(resolve, ms));
+    },
+    async updateMessages() {
```

Comment:
That's not quite what I meant, here you are delaying the request of 500ms, so making the app slower on purpose. What I was talking about was only the display of the spinner. What I wanted is to show the spinner only after 150ms (to not show a very short spinner). And to display it for minimum 300ms. So this should only touch the `isLoading` variable. But it seems too complex for now so just delete that and return to the previous state (and create an issue). We wil see about that later. Also you should no set message to [] before the request. We still want to see the data.